sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/csv/UnivocityParser.scala

```diff
@@ -317,7 +317,15 @@ class UnivocityParser(
         if (skipRow) {
           row.setNullAt(i)
         } else {
-          row(i) = valueConverters(i).apply(getToken(tokens, i))
+          // This is required to not set value as null ,
+          // 1. If the missing value at the end of line.
+          // 2. If the missing value at the beginning of line.
+          if (!options.naFilter && (i>=tokens.length ||
+            (i==0 && getToken(tokens, i).length == 0))) {
+            row(i) = valueConverters(i).apply("")
```

Comment:
For `""`, it's interpreted differently with `nullValue` or `emptyValue`. I don't think we should rely on this conversion here. Can we control this by setting one of these options?

Reply:
This is only called if the missing values in the beginning or end . In that case current code goes into exception block and update row with default values (which will replace missing values with null) . In case option.naFilter is False , we do not want to replace with null and let them be missing value